src/dotnet/commands/dotnet-restore/NuGet3.cs

```diff
@@ -19,11 +16,18 @@ public static int Restore(IEnumerable<string> args)
             }
             prefixArgs.Add("restore");
 
-            var result = new NuGetForwardingApp(Enumerable.Concat(
-                    prefixArgs,
-                    args).ToArray()).Execute();
+            var nugetApp = new NuGetForwardingApp(Enumerable.Concat(prefixArgs, args));
 
-            return result;
+            // setting NUGET_XPROJ_WRITE_TARGETS will tell nuget restore to install .props and .targets files
+            // coming from NuGet packages
+            const string nugetXProjWriteTargets = "NUGET_XPROJ_WRITE_TARGETS";
+            bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null;
+            if (setXProjWriteTargets)
+            {
+                nugetApp.WithEnvironmentVariable(nugetXProjWriteTargets, "true");
```

Comment:
The idea is that if someone else sets this to 'false', or some other value, we don't want to overwrite what they are setting.